### PR TITLE
Add Ansible for Fedora Linux

### PR DESCRIPTION
**Pull request by mbacchi**

### CODE REVIEW

File: ansible/scripts/install_kubectx.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+set -x
+
+# From https://github.com/ahmetb/kubectx#linux
+git clone https://github.com/ahmetb/kubectx.git ~/.kubectx
+
+ln -s ~/.kubectx/kubectx ~/bin/kubectx
+ln -s ~/.kubectx/kubens ~/bin/kubens
+
+cat << FOE >> ~/.bashrc
```
Comment (jkim-codecademy) [inline ansible/scripts/install_kubectx.sh:11]: should probably add assumption user is running bash

File: ansible/scripts/install_rbenv.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# set -x
```
Comment (jkim-codecademy) [inline ansible/scripts/install_rbenv.sh:3]: should this still be here

File: ansible/linux_bootstrap.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        - name: Install nvm
+          script:
+            cmd: /usr/bin/curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash
+          register: result
+
+        - name: Show debug output
+          debug:
+            var: result
+
+        - name: Install helmfile in ~/bin
```
Comment (jkim-codecademy) [inline ansible/linux_bootstrap.yml:60]: should probably install helm-diff plugin for helm

Comment (mbacchi) [reply ansible/linux_bootstrap.yml:60]: Good call. Added in new commit.

File: ansible/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Some of these tools versions are set in the `linux_bootstrap.yml` file as Ansible variables (`vars`). These can be
+modified as the `bootstrap.sh` shell script VARS are modified to be kept in line with versions pinned there.
+
+## Caveats
+
+The below caveats are listed to inform the reader about workarounds that were necessary to install these tools.
+
+* These ansible playbooks assume the user is running on Fedora 32.
+* These ansible playbooks assume the user is running the bash shell.
+* We do not install Docker currently due to the fact that Docker is not supported/functional without some
```
Comment (jkim-codecademy) [inline ansible/README.md:42]: mebe append directions to do so

File: ansible/linux_bootstrap.yml [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        - name: Check awscliv2 exists in ~/bin
+          stat: path=~/bin/aws
+          register: aws
+
+        - name: Install awscli v2 to ~/bin
+          shell: |
+            curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+            unzip awscliv2.zip
+            ./aws/install --install-dir ~/bin --bin-dir ~/bin
+            rm awscliv2.zip
```
Comment (jkim-codecademy) [inline ansible/linux_bootstrap.yml:91]: this seems to delete the zip but not the directory that was created

File: ansible/linux_bootstrap.yml [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+          dnf:
+            disable_gpg_check: yes
+            name: https://s3.amazonaws.com/session-manager-downloads/plugin/latest/linux_64bit/session-manager-plugin.rpm
+          become: yes
+
+        - name: Check terraform exists in ~/bin
+          stat: path=~/bin/terraform
+          register: terraform
+
+        - name: Install terraform v0.12.29 to ~/bin
```
Comment (jkim-codecademy) [inline ansible/linux_bootstrap.yml:120]: might as well set these versions as  vars while we're in here so bumping up veriso numbers is easier

File: ansible/linux_bootstrap.yml [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+        - name: Show debug output
+          debug:
+            var: result
+
+        - name: Check helm exists in ~/bin
+          stat: path=~/bin/helm
+          register: helm
+
+        - name: Install helm v2.14.3 to ~/bin
```
Comment (jkim-codecademy) [inline ansible/linux_bootstrap.yml:168]: versions as  vars

File: ansible/linux_bootstrap.yml [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+            rm helm-v2.14.3-linux-amd64.tar.gz
+            rm -rf linux-amd64/
+          when: not helm.stat.exists
+          register: result
+
+        - name: Check helm3 exists in ~/bin
+          stat: path=~/bin/helm3
+          register: helm3
+
+        - name: Install helm v3.3.4 to ~/bin
```
Comment (jkim-codecademy) [inline ansible/linux_bootstrap.yml:183]: versions as  vars

File: ansible/linux_bootstrap.yml [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+            path: "{{ ansible_env.HOME }}/.helm/plugins"
+            state: directory
+            mode: '0755'
+          when: not plugins.stat.exists
+
+        - name: Check helm diff exists in helm home
+          stat: path={{ ansible_env.HOME }}/.helm/plugins/diff
+          register: helmdiff
+
+        - name: Install helm-diff v2.11.0+5 to helm home
```
Comment (jkim-codecademy) [inline ansible/linux_bootstrap.yml:208]: is the versioning here correct? the "+5" seems weird. Also how does this work when installing both helm 2 ad 3

File: ansible/linux_bootstrap.yml [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+          shell: |
+            curl -L "https://github.com/databus23/helm-diff/releases/download/v2.11.0%2B5/helm-diff-linux.tgz" | tar -C $(helm home)/plugins -xzv
+            # rm helm-diff-linux.tgz
```
Comment (jkim-codecademy) [inline ansible/linux_bootstrap.yml:211]: probably should get rid of commet

File: ansible/linux_bootstrap.yml (hunk 9 above)
Comment (mbacchi) [reply ansible/linux_bootstrap.yml:208]: I will have to test, but I think I can also add helm-diff for helm3 via [these](https://github.com/databus23/helm-diff#using-helm-plugin-manager--23x) instructions using `helm plugin install https://github.com/databus23/helm-diff`.

**Review by jkim-codecademy — APPROVED** [review]
cant test since i dont run fedora but I think it looks fine,

**Pull request merged**